langchain4j-azure-open-ai/src/main/java/dev/langchain4j/model/azure/InternalAzureOpenAiHelper.java

```diff
@@ -47,28 +48,26 @@ class InternalAzureOpenAiHelper {
 
     public static final String DEFAULT_USER_AGENT = "langchain4j-azure-openai";
 
-    public static OpenAIClient setupOpenAIClient(String endpoint, String serviceVersion, String apiKey, Duration timeout, Integer maxRetries, ProxyOptions proxyOptions, boolean logRequestsAndResponses) {
+    public static <T> T setupClient(Class<T> clientType, String endpoint, String serviceVersion, Object credential, Duration timeout, Integer maxRetries, ProxyOptions proxyOptions, boolean logRequestsAndResponses) {
         OpenAIClientBuilder openAIClientBuilder = setupOpenAIClientBuilder(endpoint, serviceVersion, timeout, maxRetries, proxyOptions, logRequestsAndResponses);
 
-        return openAIClientBuilder
-                .credential(new AzureKeyCredential(apiKey))
-                .buildClient();
-    }
-
-    public static OpenAIClient setupOpenAIClient(String endpoint, String serviceVersion, KeyCredential keyCredential, Duration timeout, Integer maxRetries, ProxyOptions proxyOptions, boolean logRequestsAndResponses) {
-        OpenAIClientBuilder openAIClientBuilder = setupOpenAIClientBuilder(endpoint, serviceVersion, timeout, maxRetries, proxyOptions, logRequestsAndResponses);
-
-        return openAIClientBuilder
-                .credential(keyCredential)
-                .buildClient();
-    }
-
-    public static OpenAIClient setupOpenAIClient(String endpoint, String serviceVersion, TokenCredential tokenCredential, Duration timeout, Integer maxRetries, ProxyOptions proxyOptions, boolean logRequestsAndResponses) {
-        OpenAIClientBuilder openAIClientBuilder = setupOpenAIClientBuilder(endpoint, serviceVersion, timeout, maxRetries, proxyOptions, logRequestsAndResponses);
+        if (credential instanceof String) {
+            openAIClientBuilder.credential(new AzureKeyCredential((String) credential));
+        } else if (credential instanceof KeyCredential) {
+            openAIClientBuilder.credential((KeyCredential) credential);
+        } else if (credential instanceof TokenCredential) {
+            openAIClientBuilder.credential((TokenCredential) credential);
+        } else {
+            throw new IllegalArgumentException("Unsupported credential type: " + credential.getClass());
+        }
 
-        return openAIClientBuilder
-                .credential(tokenCredential)
-                .buildClient();
+        if (clientType == OpenAIAsyncClient.class) {
+            return clientType.cast(openAIClientBuilder.buildAsyncClient());
```

Comment:
This can be ignored: I'm having a doubt whether this would work well with a GraalVM native build

Reply:
removed reflection logic and made it GraalVM native build friendly.